### ISSUE TITLE
`upgrade` to the version that's already installed

### ISSUE THREAD

**Issue by c10l** [mas-cli/mas#501]
## Your Environment

- `mas version`: 1.8.6
- macOS version (`system_profiler SPSoftwareDataType -detailLevel mini`): 10.1...

```shell
❯ system_profiler SPSoftwareDataType -detailLevel mini
Software:

    System Software Overview:

      System Version: macOS 13.5.2 (22G91)
      Kernel Version: Darwin 22.6.0
      Time since boot: 5 days, 9 hours, 55 minutes
```

### mas Install Method

- [X] `brew install mas` (homebrew-core)
- [] [mas-cli/tap](https://github.com/mas-cli/homebrew-tap)
- [] `.pkg` installer from [releases](https://github.com/mas-cli/mas/releases)
- [] Built from source
  - Fork/branch: ? (e.g. mas-cli/main)
  - Xcode version: 10.?

## Describe the Bug

Steam Link is installed on version `1.2.0`.
`mas` sees updated version `1.3.4` available.
`mas` "upgrades" it to the old, already installed `1.2.0` version.

```shell
❯ mas outdated
1246969117 Steam Link (1.2.0 -> 1.3.4)
❯ mas upgrade
Upgrading 1 outdated application:
Steam Link (1.2.0)
```

If I uninstall and reinstall, I get `1.3.4` but running `upgrade` actually downgrades it to `1.2.0`.

```shell
❯ mas install 1246969117
==> Downloading Steam Link
==> Installed Steam Link
❯ mas info 1246969117
Steam Link 1.3.4 [0.0]
By: Valve Corporation
Released: 2023-09-12
Minimum OS: 11.0
Size: 55.8 MB
From: https://apps.apple.com/gb/app/steam-link/id1246969117?uo=4
❯ mas upgrade
Upgrading 1 outdated application:
Steam Link (1.2.0)
==> Installed Steam Link
```

## To Reproduce

1. Install Steam Link: `mas install 1246969117`
2. Run `mas upgrade`
3. Observe the downgrade <- second bug above
4. Run `mas upgrade` again
5. Observe it reinstalling the old, already installed `1.2.0` version <- first bug above

## Expected Behavior

`mas upgrade` installs Steam Link `1.3.4`, and subsequent upgrades do nothing as it's already latest.

**Comment by Tekl** [comment]
Looks like this is a duplicate of #491 and there's a potential fix at #496.

**Comment by kishaningithub** [comment]
The same is happening to me for [endel mac app](https://endel.io/). Simulation steps below.

```bash
$ mas list
1346247457  Endel  (4.31.12)

$ mas outdated
1346247457 Endel (4.31.12 -> 4.38.2)

$ mas upgrade
Upgrading 1 outdated application:
Endel (4.31.12)
==> Installed Endel: Focus, Sleep, Relax

$ mas outdated
1346247457 Endel (4.31.12 -> 4.38.2)
```

**Comment by jun6lee** [comment]
I see the same with Omnivore (1.38.0)

> mas outdated
1564031042 Omnivore (1.38.0 -> 1.41)

**Comment by rosstimson** [comment]
I am getting this same issue for Pomodoro timer app Flow:

```
x mas list | rg Flow
1423210932  Flow                           (2.14.1)

$ mas outdated
1423210932 Flow (2.14.1 -> 3.3.1)
```

After reading through https://github.com/mas-cli/mas/pull/496 mentioned above I've confirmed that version `3.3.1` here is indeed the latest iOS version, the version I have 2.14.1 is the latest macOS version.

**Comment by rgoldberg** [comment]
Closing as a duplicate of #491. PR #496 fixed the issue. It will be released in the upcoming version 1.8.7, which is currently in beta.

If you think this issue is different than #491, please let us know.
